bin/create_folders.py

```diff
@@ -226,11 +264,14 @@ def create_folders(
     project_name = sys.argv[1]
     submission_id = sys.argv[2]
     create_or_update = sys.argv[3]
-    predictions_file = sys.argv[4] if len(sys.argv) > 4 else None
+    # TODO: https://sagebionetworks.jira.com/browse/ORCA-311
+    predictions_file = sys.argv[4] if len(sys.argv) == 5 else None
```

Comment:
If I understand this logic your either passing in a:
`predictions_file`
or
`log_file`
?

I just wanted to double check on the intentionality behind these `len` checks.

Reply:
The `len()` checks are to ensure the variables `predictions_file` and `log_file` get assigned to the right input variable(s) depending on how many inputs were fed in the command line call to `create_folders.py`. If we have 5 input calls, we can assume `predictions_file` is the only thing fed. If we have 6, `predictions_file` and `log_file` are both fed.

However the order is strict (you can't feed `log_file` without feeding `predictions_file` or at least declaring it as None), so that's why I plan to implement argparse so we have specific args assigned to inputs. For now this will not cause any problems, since the model-to-data infrastructure takes this into account.

Reply:
Gotcha. Based on the logic here wouldn't this be a problem though?

length = 5
0. Something
1. project_name
2. submission_name
3. create_or_update
4. predictions_file
predictions_file is filled in from sys.argv[4]
log_file is not because len(sys.argv) == 5



length = 6
0. Something
1. project_name
2. submission_name
3. create_or_update
4. predictions_file
5. log_file 
predictions_file is NOT filled in because len(sys.argv) != 5
log_file is filled in because len(sys.argv) == 6


In this second case it would seem to me that you are dropping the `predictions_file`.

I'm thinking that these length checks should instead have been `>=` instead of `==`

Reply:
You're right, this logic sets `predictions_file` to None if `log_file` is also fed, so this logic should instead be `if len(sys.argv) >= 5 else None`. Thanks for catching this!